test/integration/targets/win_psrepository/tasks/test.yml

```diff
@@ -0,0 +1,140 @@
+# This file is part of Ansible
+
+# Copyright: (c) 2018, Wojciech Sciesinski <wojciech[at]sciesinski[dot]net>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+---
+
+- name: install updated NuGet version
+  win_shell: 'Install-PackageProvider -Name NuGet -MinimumVersion 2.8.5.201 -Force'
+
+- name: check setting of InstallationPolicy
+  win_psrepository:
+    name: PSGallery
+    source_location: "{{ PSGallery_url }}"
+    installation_policy: trusted
+  register: change_installation_policy_1
+
+- name: get result of setting of InstallationPolicy
+  win_shell: '(Get-PSRepository -Name PSGallery).InstallationPolicy'
+  changed_when: false
+  register: result_change_installation_policy_1
+
+- name: test check setting of InstallationPolicy
+  assert:
+    that:
+      - "change_installation_policy_1 is changed"
+      - "'{{ result_change_installation_policy_1.stdout | trim }}' == 'Trusted'"
+
+- name: check idempotency setting of InstallationPolicy
+  win_psrepository:
+    name: PSGallery
+    source_location: "{{ PSGallery_url }}"
+    installation_policy: trusted
+  register: change_installation_policy_2
+
+- name: test idempotency setting of InstallationPolicy
+  assert:
+    that:
+      - "change_installation_policy_2 is not changed"
+
+- name: check adding of repository
+  win_psrepository:
+    name: "{{ repository_name }}"
+    source_location: "{{ repository_sourcelocation }}"
+    state: present
+  register: adding_repository_1
+
+- name: get result of adding of repository 1 - 1
```

Comment:
Instead of having 3 separate tasks to get the results, just use 1 win_shell task to get them all, e.g.

```
- name: get result of adding repository
  win_shell: |
    $repo = Get-PSRepository -Name {{ repository_name|quote }} -ErrorAction Stop
    $repo.SourceLocation
    $repo.InstallationPolicy
  register: result_adding_repository_1
```

This way you can access the source with `{{ result_adding_repository_1.stdout_lines[0] }}` and the policy with `{{ result_adding_repository_1.stdout_lines[1] }}`.